client/me/notification-settings/wpcom-settings/index.jsx

```diff
@@ -45,103 +46,82 @@ const options = {
 class WPCOMNotifications extends React.Component {
 	static displayName = 'WPCOMNotifications';
 
-	state = {
-		settings: null,
-	};
+	// TODO: Add propTypes
 
 	componentDidMount() {
-		store.on( 'change', this.onChange );
-		fetchSettings();
+		this.props.fetchSettings();
```

Comment:
There is a convention to prefer a separate `<QueryNotificationsSettings />` component instead of directly calling the `fetchSettings` action in a lifecycle method.

Reply:
I thought about doing this but it feels like work for another PR - I worry that the domino effect of changing one file that requires another file to be changed that requires another file to be changed could get out of hand here - there's a lot of work that could be done and it'd be a shame to let that stop us merging improvements, IMO